with-camera/App.js

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect } from 'react';
+import { Text, View, TouchableOpacity, ImageBackground, Platform } from 'react-native';
+import { Camera } from 'expo-camera';
+
+export default function App() {
+  const [hasPermission, setHasPermission] = useState(null);
```

Comment:
Split the permission into a new component that guards the custom camera component.